### PR TITLE
Yarn dependencies upgrade & yarn lint update

### PR DESCRIPTION
**Pull request by davidmfinol**
#### Changes

- Remove unneeded `git add` from pre-commit hook
- Update dependencies to latest versions for non `eslint` dependencies
- Switch from `npx lint-staged` to `yarn lint-staged` in pre-commit hook
- Add `yarn lint` to pre-commit hook
- Run updated version of `prettier`
- Temporarily ignore `eslint` rule since full `eslint` update and refactor will be done in a follow-up PR

#### Checklist

- [x] Read the contribution [guide](../CONTRIBUTING.md) and accept the [code](../CODE_OF_CONDUCT.md) of conduct
- [x] Readme (updated or not needed)
- [x] Tests (added, updated or not needed)

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -1,7 +1,10 @@
 #!/bin/sh
 . "$(dirname "$0")/_/husky.sh"
 
-npx lint-staged
+if [ -t 1 ]; then
+  exec < /dev/tty
+fi
```
[the remark this answers is not in the document]
Comment (davidmfinol) [reply .husky/pre-commit]: Husky was intermittently throwing errors during the pre-commit hook, and this was something I found that helped me avoid the errors.
However, the better solution is to simply remove `git add` from the pre-commit hook, so I've done that, and I've removed this code too.

[the remark this answers is not in the document]
Comment (webbertakken) [reply .husky/pre-commit]: Perfect!

Indeed git add is now implicit to lint-staged tasks so like this is looks correct.

**Comment by webbertakken** [comment]
I see many rules from the updates linting configs aren't yet accounted for. Do you think it makes sense to leave the linting configs at their current versions in this PR and optionally make a separate PR for adapting the repo to the new rules?

I'm asking because in my experience linting rules usually make not just for small code changes but possibly also restructuring of code, filenames etc.

**Comment by davidmfinol** [comment]
I was thinking of spending some time and doing the refactor with this PR, but you're right: It's probably better to do it in a separate PR.
I'll see if I can keep the current version of eslint while still updating everything else.

**Comment by davidmfinol** [comment]
I believe this PR is now ready to be merged. I'll follow-up with the `eslint` refactor PR next week.

**Pull request merged**